tests/test_camera.py

```diff
@@ -53,3 +57,61 @@ def read(self):
             video_source = VideoSource(camera_id=0)
         assert video_source._frames is None
         assert video_source.get_image() is not None
+
+
+class TestVideoStream(unittest.TestCase):
+
+    def setUp(self) -> None:
+        self.video = VideoSource(filename=VIDEO_PATH)
+        self.stream = VideoStream(video_source=self.video, fps=12.0)
+
+    def test_stop(self):
+        self.stream.stop()
+        self.assertTrue(self.stream._shutdown)
+
+    def test_extract_image(self):
+        img_tuple = self.video.get_image()
+        self.stream.frames.put(img_tuple, False)
+        imgs = self.stream.get_image()
+        assert type(imgs[0]) == np.ndarray
+
+    def test_frame_conditions(self):
+        self.stream.run()
+        self.assertTrue(self.stream.frames.full())
+        self.assertTrue(self.stream._shutdown)
+
+
+class TestVideoWriter(unittest.TestCase):
+
+    def setUp(self) -> None:
+        self.output_video_path = os.path.join(os.path.dirname(__file__), 'resources', 'test_writer.mp4')
+        self.videowriter = VideoWriter(path=self.output_video_path, fps=12.0, resolution=(40, 30))
+
+    def test_write(self):
+        input_video = cv2.VideoCapture(VIDEO_PATH)
+
+        while True:
+            ret, frame = input_video.read()
+            self.videowriter.write(frame)
+            if not ret:
+                break
+
+        input_video.release()
+        self.videowriter.release()
+        output_video = cv2.VideoCapture(self.output_video_path)
+
+        # Check whether new output file is the same as the original
+        while True:
+            ret, frame_in = input_video.read()
+            _, frame_out = output_video.read()
```

Comment:
As you have committed the output video to the resources folder, this test case would also work without any new output video being written. It would be cleaner to not commit the video and always remove it after it has been written in this test case.